packages/react-components/src/components/timegraph-output-component.tsx

```diff
@@ -169,10 +172,21 @@ export class TimegraphOutputComponent extends AbstractTreeOutputComponent<Timegr
             styleModel: await this.styleProvider.getStyleModel()
         });
         this.waitAnalysisCompletion();
+        this.subscribeToEvents();
     }
 
     componentWillUnmount(): void {
         super.componentWillUnmount();
+        this.unsubscribeToEvents();
+    }
+
+    protected subscribeToEvents(): void {
+        signalManager().on(Signals.THEME_CHANGED, this.onThemeChange);
+        signalManager().on(Signals.SELECTION_CHANGED, this.onSelectionChanged);
+    }
+
+    protected unsubscribeToEvents(): void {
+        signalManager().off(Signals.THEME_CHANGED, this.onThemeChange);
```

Comment:
Small refactor: Created subscribeToEvents and unsubToEvents methods since I added my own signals.